packages/components/src/DateField/DateInput.tsx

```diff
@@ -0,0 +1,54 @@
+import { ReactElement } from 'react';
+import { DateInput, Group } from 'react-aria-components';
+import type RAC from 'react-aria-components';
+
+import { cn, useClassNames } from '@marigold/system';
+
+import { DateSegment } from './DateSegment';
+
+type RemovedProps = 'style' | 'className' | 'children';
+
+export interface DateInputProps extends Omit<RAC.DateInputProps, RemovedProps> {
+  variant?: string;
+  size?: string;
+  action?: ReactElement;
+}
+
+const _DateInput = ({ variant, size, action, ...props }: DateInputProps) => {
+  const classNames = useClassNames({ component: 'DateField', variant, size });
+
+  return (
+    <Group className={cn('flex flex-1 items-center', classNames.field)}>
+      <DateInput className={cn('flex flex-1 items-center')} {...props}>
+        {segment => (
+          <DateSegment
+            className={cn(
+              'group/segment',
```

Comment:
classnames.segment is only used here. So we can put the additional also in segment?